utils/fakes.py

```diff
@@ -0,0 +1,39 @@
+class FakeRedis(object):
+    """ Essentially just a dictionary accessed via 'get' and 'set' methods """
+
+    def __init__(self):
+        self.storage = {}
+        self.ex = None
+
+    def set(self, key, value, ex=0):
+        self.storage[key] = str(value)
+        self.ex = ex
+
+    def get(self, key, ex=0):
+        self.ex = ex
+        return self.storage.get(key, None)
+
+
+class FakeS3(object):
+    """ Essentially just a dictionary where the keys are tuples of (bucket_name, path) """
```

Comment:
Do you mean where is the data written to? Or where are things using FakeS3? Former: just into memory. Latter: in this PR it's only used by the source API test. 
This mostly needs to exist because the docker-based fake S3 isn't available to travis.